src/sage/combinat/root_system/weyl_group.py

```diff
@@ -528,15 +536,15 @@ def long_element_hardcoded(self):
                 m = matrix(QQ, 8, l)
             else:
                 raise NotImplementedError("not implemented yet for this type")
-        elif type[0] == 'G':
-            third = ZZ(1) / ZZ(3)
-            twothirds = ZZ(2) / ZZ(3)
+        elif typ[0] == 'G':
+            third = QQ((1, 3))
+            twothirds = QQ((2, 3))
             l = [[-third, twothirds, twothirds],
                  [twothirds, -third, twothirds],
                  [twothirds, twothirds, -third]]
             m = matrix(QQ, 3, l)
         else:
-            m = diagonal_matrix([-1 for i in range(self.n)])
+            m = diagonal_matrix([-1 for _ in range(self.n)])
```

Comment:
```suggestion
            m = diagonal_matrix([-1]*self.n)
```
since we aren't mutating anything.